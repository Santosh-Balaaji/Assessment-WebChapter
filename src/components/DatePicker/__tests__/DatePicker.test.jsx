import React from 'react';
import { shallow } from 'enzyme';
import DatepickerComponent from '../DatePicker';

describe('DatePicker Component', () => {
  let wrapper;
  const childrenName = 'Submit';
  const mockFnOnChange = jest.fn();
  const mockFnOnBlur = jest.fn();
  const stubProps = {
    name: 'ExpiryDate',
    id: 'expirydate',
    value: null,
    onChange: mockFnOnChange,
    onBlur: mockFnOnBlur,
    placeholder: '',
    label: '',
  };

  beforeAll(() => {
    wrapper = shallow(<DatepickerComponent {...stubProps} />);
  });

  it('should render default component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange function on change of Event', () => {
    wrapper
      .find('PickerWithState[data-testid="keyboardDatePicker"]')
      .simulate('change', { target: { name: 'Sample', value: '12/12/2020' } });
    expect(mockFnOnChange).toHaveBeenCalled();
  });

  it('should call handleBlur function on Blur', () => {
    wrapper
      .find('PickerWithState[data-testid="keyboardDatePicker"]')
      .simulate('blur');
    expect(mockFnOnBlur).toHaveBeenCalled();
  });
});
