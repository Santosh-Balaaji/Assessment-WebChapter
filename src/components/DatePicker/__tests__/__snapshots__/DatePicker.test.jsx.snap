// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Component should render default component 1`] = `
<Fragment>
  <MuiPickersUtilsProvider
    utils={[Function]}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1 datepicker"
      container={true}
      data-id="date-grid-container"
      justify="space-between"
    >
      <PickerWithState
        KeyboardButtonProps={
          Object {
            "aria-label": "change date",
          }
        }
        TextFieldComponent={[Function]}
        allowKeyboardControl={true}
        autoOk={true}
        data-testid="keyboardDatePicker"
        disableToolbar={true}
        format="DD-MM-YYYY"
        id="expirydate"
        inputVariant="outlined"
        invalidDateMessage="Invalid Date Format"
        label=""
        maxDate={2100-01-01T00:00:00.000Z}
        maxDateMessage="Date should not be after maximal date"
        minDate={1900-01-01T00:00:00.000Z}
        minDateMessage="Date should not be before minimal date"
        onBlur={[MockFunction]}
        onChange={[Function]}
        openTo="date"
        value={null}
        variant="inline"
        views={
          Array [
            "year",
            "date",
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </MuiPickersUtilsProvider>
</Fragment>
`;
